### ISSUE TITLE
Performance testing of Fergus SSL

### ISSUE THREAD

**Issue by magsol** [quinngroup/fergus-ssl#1]
Need a performance benchpark for the current algorithmic framework. Use `sklearn.datasets` to conduct semi-supervised learning testing, and time the results for varying dataset sizes as well as accuracy.

**Comment by magsol** [comment]
We still need timings for the dataset.
- How long does it take to run?
- How does changing the size of the dataset change the runtime?
- Where are the bottlenecks?
- How does varying `gamma` affect the clustering quality?
- How does adding noise to the half-moons affect the clustering quality?
